unity-renderer/Assets/UIComponents/Tests/ButtonComponentViewTests.cs

```diff
@@ -0,0 +1,118 @@
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ButtonComponentViewTests
+{
+    private ButtonComponentView buttonComponent;
+
+    [SetUp]
+    public void SetUp() { buttonComponent = CreateButtonComponent(); }
+
+    [TearDown]
+    public void TearDown()
+    {
+        buttonComponent.Dispose();
+        GameObject.Destroy(buttonComponent.gameObject);
+    }
+
+    [Test]
+    public void SetOnClickCorrectly()
+    {
+        // Arrange
+        bool isClicked = false;
+        buttonComponent.onClick.AddListener(() => isClicked = true);
+
+        // Act
+        buttonComponent.button.onClick.Invoke();
+
+        // Assert
+        Assert.IsTrue(isClicked, "The button has not responded to the onClick action.");
+    }
+
+    [Test]
+    public void ConfigureButtonCorrectly()
+    {
+        // Arrange
+        ButtonComponentModel testModel = new ButtonComponentModel
+        {
+            icon = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero),
```

Comment:
The `Texture2D` objects have to be destroyed manually to avoid leaks.

Reply:
Good catch! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/531831692da512ed2d0095f4e2c0d08be70ee7be